include/mount-api-utils.h

```diff
@@ -203,6 +209,133 @@ static inline int fspick(int dfd, const char *pathname, unsigned int flags)
+	uint32_t mnt_id_old;	/* Reused IDs used in proc/.../mountinfo */
+	uint32_t mnt_parent_id_old;
+	uint64_t mnt_attr;		/* MOUNT_ATTR_... */
+	uint64_t mnt_propagation;	/* MS_{SHARED,SLAVE,PRIVATE,UNBINDABLE} */
+	uint64_t mnt_peer_group;	/* ID of shared peer group */
+	uint64_t mnt_master;	/* Mount receives propagation from this ID */
+	uint64_t propagate_from;	/* Propagation from in current namespace */
+	uint32_t mnt_root;		/* [str] Root of mount relative to root of fs */
+	uint32_t mnt_point;	/* [str] Mountpoint relative to current root */
+	uint64_t __spare2[50];
+	char str[];		/* Variable size part containing strings */
+};
+#endif
+
+/* sb_flags (defined in kernel include/linux/fs.h) */
+#ifndef SB_RDONLY
+# define SB_RDONLY       BIT(0)	/* Mount read-only */
+# define SB_NOSUID       BIT(1)	/* Ignore suid and sgid bits */
+# define SB_NODEV        BIT(2)	/* Disallow access to device special files */
+# define SB_NOEXEC       BIT(3)	/* Disallow program execution */
+# define SB_SYNCHRONOUS  BIT(4)	/* Writes are synced at once */
+# define SB_MANDLOCK     BIT(6)	/* Allow mandatory locks on an FS */
+# define SB_DIRSYNC      BIT(7)	/* Directory modifications are synchronous */
+# define SB_NOATIME      BIT(10)	/* Do not update access times. */
+# define SB_NODIRATIME   BIT(11)	/* Do not update directory access times */
+# define SB_SILENT       BIT(15)
+# define SB_POSIXACL     BIT(16)	/* Supports POSIX ACLs */
+# define SB_INLINECRYPT  BIT(17)	/* Use blk-crypto for encrypted files */
+# define SB_KERNMOUNT    BIT(22)	/* this is a kern_mount call */
+# define SB_I_VERSION    BIT(23)	/* Update inode I_version field */
+# define SB_LAZYTIME     BIT(25)	/* Update the on-disk [acm]times lazily */
+#endif
+
+/*
+ * @mask bits for statmount(2)
+ */
+#ifndef STATMOUNT_SB_BASIC
+# define STATMOUNT_SB_BASIC		0x00000001U     /* Want/got sb_... */
+#endif
+#ifndef STATMOUNT_MNT_BASIC
+# define STATMOUNT_MNT_BASIC		0x00000002U	/* Want/got mnt_... */
+#endif
+#ifndef STATMOUNT_PROPAGATE_FROM
+# define STATMOUNT_PROPAGATE_FROM	0x00000004U	/* Want/got propagate_from */
+#endif
+#ifndef STATMOUNT_MNT_ROOT
+# define STATMOUNT_MNT_ROOT		0x00000008U	/* Want/got mnt_root  */
+#endif
+#ifndef STATMOUNT_MNT_POINT
+# define STATMOUNT_MNT_POINT		0x00000010U	/* Want/got mnt_point */
+#endif
+#ifndef STATMOUNT_FS_TYPE
+# define STATMOUNT_FS_TYPE		0x00000020U	/* Want/got fs_type */
+#endif
+
+/*
+ * Special @mnt_id values that can be passed to listmount
+ */
+#ifdef LSMT_ROOT
+# define LSMT_ROOT		0xffffffffffffffff	/* root mount */
+#endif
+
+#ifndef LISTMOUNT_REVERS
+# define LISTMOUNT_REVERSE      (1 << 0) /* List later mounts first */
+#endif
+
+#if !defined(HAVE_STATMOUNT) && defined(SYS_statmount)
+static inline int statmount(uint64_t mnt_id, uint64_t mask,
+			struct statmount *buf,
+			size_t bufsize, unsigned int flags)
+{
+       struct mnt_id_req req = {
+               .size = MNT_ID_REQ_SIZE_VER0,
```

Comment:
I'm not sure. Technically, you're right, but kernel headers define the _VER* macros, and I think it's better to use them for readability. Kernel developers also use them in all examples and tests.

I think it's better to interpret the field as "I want VER0" functionality rather than "I use VER0" request struct. There is already VER1 in 6.10.0-rc1, and it will be possible to keep only one `struct mnt_id_req` definition in the code and use it for VER0 and VER1 according to the desired operation (VER1 introduces namespaces).

Reply:
Ack